### PR TITLE
[All] Added TintColor Property To Image

### PR DESCRIPTION
**Pull request by ravinderjangra**
### Description of Change ###
Implementation of https://forums.xamarin.com/discussion/84674/add-tint-property-to-image
Added new attachedproperty to Image to use TintColor in Xamain.Forms

### API Changes ###

var image = new Image();
image.TintColor = Color.Red; //new Property

Added:
 - Color TintColorProperty { get; set; } //BindableProperty 
 - void SetTint() // Android Image Renderer (Platform)
 - void SetTintColor()//ios Image Renderer (Platform)

### PR Checklist ###

- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense
screenshot 34

### CODE REVIEW

File: Xamarin.Forms.Core/Image.cs [hunk 1]
```diff
@@ -19,6 +19,8 @@ public class Image : View, IImageController, IElementConfiguration<Image>
 
 		internal static readonly BindablePropertyKey IsLoadingPropertyKey = BindableProperty.CreateReadOnly("IsLoading", typeof(bool), typeof(Image), default(bool));
 
+		public static readonly BindableProperty TintColorProperty = BindableProperty.Create("TintColor", typeof(Color), typeof(Color), Color.Transparent);
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Image.cs]: - use `nameof(TintColor)` instead of `"TintColor"`
- use `Color.Default` or `default(Color)` as default color

**Comment by ravinderjangra** [comment]
Okay. I will try to achieve on UWP too.

**Comment by ravinderjangra** [comment]
Hi, I was searching for the solution to add the tintcolor property for uwp too, But with the older version, it is not possible without using 3rd party library. either I will have to upgrade the Microsoft.NETCore.UniversalWindowsPlatform or use some other libraries. 

Can't we just go with Android and iOS only?

**Comment by andreinitescu** [comment]
@ravinderjangra Maybe do this https://stackoverflow.com/questions/38199867/uwp-apply-colorfilter-to-bitmapimage

**Comment by ravinderjangra** [comment]
@andreinitescu @StephaneDelcroix after searching more I found that in UWP all these effects and image processing is done using Win2D (https://www.nuget.org/packages/Win2D.uwp) so if using this is no issue then it can be done easily.

**Comment by StephaneDelcroix** [comment]
@ravinderjangra we'll come back to you on what to do next or if it's good as is

**Comment by jassmith** [comment]
Following up with management on this not working on UWP and still merging. Will come back in a couple days with an answer

**Comment by ravinderjangra** [comment]
Any Update on this?

**Comment by ravinderjangra** [comment]
Still no update?

**Comment by Happypig375** [comment]
I'll bump this PR too.

**Comment by rmarinho** [comment]
@ravinderjangra now that we are on netstandard 2.0 can you take a look at a UWP implementation ? 

Thanks

**Comment by ravinderjangra** [comment]
@rmarinho Microsoft.UI.Composition and Microsoft.Graphics.Canvas is still not part of .net standard 2.0.
checked at: https://docs.microsoft.com/en-us/dotnet/api/?view=netstandard-2.0&term=Microsoft.UI.Composition
So nothing can be done without adding the nuget https://www.nuget.org/packages/Win2D.uwp.
Reference : http://microsoft.github.io/Win2D/html/Introduction.htm

I can go with adding nuget that can be used in future for animation and effect purpose.

**Comment by rookiejava** [comment]
In tizen backend, there is same TizenSpecific API as below.

https://github.com/xamarin/Xamarin.Forms/blob/a24c7e5808413c6e59a0affc4ba7f2346d105e71/Xamarin.Forms.Core/PlatformConfiguration/TizenSpecific/Image.cs#L7

So, you can easily implement `Image.TintColor` with existing ImageRenderer implementation.

https://github.com/xamarin/Xamarin.Forms/blob/a24c7e5808413c6e59a0affc4ba7f2346d105e71/Xamarin.Forms.Platform.Tizen/Renderers/ImageRenderer.cs#L85

Could you add it for Tizen? Or,if you have any problem working on tizen, I'll do it with other PR. no problem.

**Comment by ravinderjangra** [comment]
@rookiejava why do we have to add this to tizenspecific code while we are doing this for all platforms so this only changes needed will be made only in imagerenderer for tizen

**Comment by rookiejava** [comment]
@ravinderjangra I think there is a misunderstanding. What I mean that is you can easily implentement `Image.TintColor` in ImageRenerer of XF.Platform.Tizen refer to the BlendColor imlementation. You don;t need to add `TintColor` to TizenSpecifics.

**Comment by ravinderjangra** [comment]
okay. thanks yes I'm going to use the same function but with transparent color check only

File: Xamarin.Forms.Platform.Android/Renderers/ImageRenderer.cs [hunk 2]
```diff
@@ -120,5 +124,25 @@ public override bool OnTouchEvent(MotionEvent e)
 
 			return _motionEventHelper.HandleMotionEvent(Parent, e);
 		}
+
+		void SetTint()
```
Comment (rookiejava) [inline Xamarin.Forms.Platform.Android/Renderers/ImageRenderer.cs]: `UpdateTintColor()` would be better for the sake of consistency.

File: Xamarin.Forms.Platform.Android/Renderers/ImageRenderer.cs [hunk 3]
```diff
@@ -120,5 +124,25 @@ public override bool OnTouchEvent(MotionEvent e)
...
+		{
+			if (Control == null || Element == null)
+				return;
+
+			if (Element.TintColor.Equals(Xamarin.Forms.Color.Transparent))
```
Comment (rookiejava) [inline Xamarin.Forms.Platform.Android/Renderers/ImageRenderer.cs]: The condition is not different between Android and iOS renderer. Keep consistency. 
```
if( Element.TintColor == Color.Transparent)
{
}
else
{
}
```

**Comment by ravinderjangra** [comment]
@rookiejava any action on this?

Because by mistake, I created this PR using my master and now I want to create some new PR so If you can merge these changes then it is okay otherwise I will have to close this pr and create a new one with from new branch.

**Comment by rookiejava** [comment]
I don't have any power to merge it. It's up to Xamarin team.

**Comment by ravinderjangra** [comment]
@StephaneDelcroix Can you merge this? If it is okay. As I already explained 

> Because by mistake, I created this PR using my master and now I want to create some new PR so If you can merge these changes then it is okay otherwise I will have to close this pr and create a new one with from new branch.

**Comment by ravinderjangra** [comment]
Any update ?

**Comment by rmarinho** [comment]
Hi @ravinderjangra after talking with the team we decided we will accept this as a PlatformSpecific or PlatformEffect, please reopen with that aproach.

**Comment by opcodewriter** [comment]
So sad. Someone started a PR, got approved initially apparently, and then it got closed by Xamarin asking for a different implementation.

**Pull request closed without merging**